### PR TITLE
only return root spans for jaeger http api

### PR DESCRIPTION
**Pull request by trinity-1686a**
### Description

on jaeger http api, return only root span. On grpc api, return all, otherwise jaeger shows empty traces until you reload the page
Can be improved once #5308 is merged, to stay mostly backward compatible (right now, the http api errors out, if `otel-traces-v0_7` exists)

### How was this PR tested?

tested manually with jaeger, and grafana jaeger datasource pointed to quickwit

### CODE REVIEW

**Comment by rdettai** [comment]
might be of interest to @Sh4d1 (see https://github.com/quickwit-oss/quickwit/issues/5122)

**Review by fmassot — APPROVED** [review]

**Comment by rdettai** [comment]
@trinity-1686a quick thought regarding your bump of OTEL_TRACES_INDEX_ID to `otel-traces-v0_9`. Now that mappings can be updated, should we not set the the index id to `otel-traces` ?

File: quickwit/quickwit-serve/src/jaeger_api/rest_handler.rs [hunk 1]
```diff
@@ -353,7 +354,7 @@ mod tests {
     async fn test_when_jaeger_not_found() {
         let jaeger_api_handler = jaeger_api_handlers(None).recover(crate::rest::recover_fn_final);
         let resp = warp::test::request()
-            .path("/otel-traces-v0_7/jaeger/api/services")
+            .path("/otel-traces-v0_9/jaeger/api/services")
```
[the remark this answers is not in the document]
Comment (rdettai) [reply quickwit/quickwit-serve/src/jaeger_api/rest_handler.rs:357]: and here https://github.com/quickwit-oss/quickwit/blob/543c84ca2cf98f62dc1dbc5a362cdbb33088d3d0/quickwit/rest-api-tests/scenarii/es_compatibility/0021-cat-indices.yaml#L25

[the remark this answers is not in the document]
Comment (trinity-1686a) [reply quickwit/quickwit-serve/src/jaeger_api/rest_handler.rs:357]: this is already fixed in 7184ccd

**Pull request merged**